src/services/upload/livePhotoService.ts

```diff
@@ -0,0 +1,162 @@
+import { FILE_TYPE } from 'constants/file';
+import { encodeMotionPhoto } from 'services/motionPhotoService';
+import {
+    FileTypeInfo,
+    FileWithCollection,
+    isDataStream,
+    LivePhotoAssets,
+    Metadata,
+} from 'types/upload';
+import { splitFilenameAndExtension } from 'utils/file';
+import { readFile } from './fileService';
+import uploadService from './uploadService';
+import UploadService from './uploadService';
+
+export function getLivePhotoFileType(
+    file1TypeInfo: FileTypeInfo,
+    file2TypeInfo: FileTypeInfo
+) {
+    return {
+        fileType: FILE_TYPE.LIVE_PHOTO,
+        exactType: `${file1TypeInfo.exactType}+${file2TypeInfo.exactType}`,
+    };
+}
+
+export function getLivePhotoMetadata(
+    file1Metadata: Metadata,
+    file2Metadata: Metadata
+) {
+    return {
+        ...file1Metadata,
+        ...file2Metadata,
+        title: `${splitFilenameAndExtension(file1Metadata.title)[0]}.zip`,
+        fileType: FILE_TYPE.LIVE_PHOTO,
+    };
+}
+
+export function getLivePhotoSize(livePhotoAssets: LivePhotoAssets) {
+    return livePhotoAssets.image.size + livePhotoAssets.video.size;
+}
+
+export async function readLivePhoto(
+    worker,
+    reader: FileReader,
+    fileTypeInfo: FileTypeInfo,
+    livePhotoAssets: LivePhotoAssets
+) {
+    const image = await readFile(
+        worker,
+        reader,
+        { exactType: fileTypeInfo.exactType, fileType: FILE_TYPE.IMAGE },
+        livePhotoAssets.image
+    );
+    const video = await readFile(
+        worker,
+        reader,
+        { exactType: fileTypeInfo.exactType, fileType: FILE_TYPE.VIDEO },
+        livePhotoAssets.video
+    );
+
+    if (isDataStream(video.filedata) || isDataStream(image.filedata)) {
+        throw new Error('too large live photo assets');
+    }
+    return {
+        filedata: await encodeMotionPhoto({
+            image: image.filedata as Uint8Array,
+            video: video.filedata as Uint8Array,
+            imageNameTitle: livePhotoAssets.image.name,
+            videoNameTitle: livePhotoAssets.video.name,
+        }),
+        thumbnail: video.hasStaticThumbnail ? video.thumbnail : image.thumbnail,
```

Comment:
Should we avoid generating or in fact using the thumbnail for video? Ideally, the thumbnail should be the same as the image of the live photo.